components/NavigationDrawer/examples/BottomDrawerExample.swift

```diff
@@ -0,0 +1,353 @@
+    super.init(coder: aDecoder)
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    view.backgroundColor = UIColor.purple
+    self.tableView.register(UITableViewCell.self, forCellReuseIdentifier: "Cell")
+    self.tableView.isScrollEnabled = false
+  }
+
+  override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+    let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath)
+    cell.textLabel?.text = "cell #\(indexPath.item)"
+    cell.backgroundColor = .white
+    print(cell.textLabel?.text ?? "")
+    return cell
+  }
+
+  override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+    return 100
+  }
+
+  override func numberOfSections(in tableView: UITableView) -> Int {
+    return 1
+  }
+}
+
+class DrawerContentWithScrollViewController: UIViewController,
+    UICollectionViewDelegate, UICollectionViewDataSource {
+
+  let collectionView: UICollectionView
+  let layout = UICollectionViewFlowLayout()
+  override var preferredContentSize: CGSize {
+    get {
+      return CGSize(width: view.bounds.width,
+                    height: collectionView.collectionViewLayout.collectionViewContentSize.height)
+    }
+    set {
+      super.preferredContentSize = newValue
+    }
+  }
+
+  init() {
+    collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+    super.init(nibName: nil, bundle: nil)
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+    super.init(coder: aDecoder)
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    self.view.backgroundColor = .lightGray
+    collectionView.frame = CGRect(x: 0, y: 0, width: self.view.bounds.width,
+                             height: self.view.bounds.height)
+    collectionView.register(UICollectionViewCell.self, forCellWithReuseIdentifier: "Cell")
+    collectionView.isScrollEnabled = false
+    collectionView.delegate = self
+    collectionView.dataSource = self
+    layout.minimumLineSpacing = 0
+    layout.minimumInteritemSpacing = 0
+    self.view.addSubview(collectionView)
+  }
+
+  override func viewWillLayoutSubviews() {
+    let s = self.view.frame.size.width / 3
+    layout.itemSize = CGSize(width: s, height: s)
+  }
+
+  func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+    return 102
+  }
+
+  func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+    let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "Cell", for: indexPath)
+    let color = CGFloat(indexPath.row % 2) * 0.2 + 0.8
+    print(indexPath.item)
+    cell.backgroundColor = UIColor(white: color, alpha: 1)
+    return cell
+  }
+
+  func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
```

Comment:
Is this required? It doesn't look like this class is a `UITableViewDataSource`.

Reply:
done, thanks for the catch